### PR TITLE
Chunk backtest fetch by month to avoid 40K pagination truncation

### LINKED ISSUE #459: Bug: Backtest truncates high-volume series at 40K pagination limit
## Problem

The backtest engine hits the 200-page pagination limit (40,000 markets) on high-volume series like KXINX (S&P 500 hourly) and KXNASDAQ100 (Nasdaq 100 hourly), even with date filtering (#410) in place.

### Evidence from a 3-month backtest (2026-01-01 to 2026-04-01):
```
Pagination limit reached (200 pages, 40000 markets)
Pagination limit reached (200 pages, 40000 markets)
```

- Total markets fetched: 84,430
- Two series hit the 40K cap = ~80,000 of the total
- Despite truncation, financials produced 66% of all trades (141/214)
- The true opportunity set is larger than what the backtest reports

Hourly contract series generate ~480 markets/day (24 hours × ~20 strike levels). Over 3 months that's ~43,000+ markets per series — exceeding the 40K cap.

## Impact

- Backtest **underestimates** the strategy's performance in financials
- Win rate, P&L, and Sharpe are based on incomplete data
- Cannot reliably compare strategy parameters when the underlying data is truncated

## Proposed Fix

**Chunk the date range** by month (or week for hourly series) so each API call stays well under the pagination limit:

```python
# Instead of one call per series for the full date range:
for series in series_list:
    markets = await list_all_markets(client, series_ticker=series,
                                     min_close_ts=start, max_close_ts=end,
                                     max_pages=200)

# Break into monthly chunks:
for series in series_list:
    for chunk_start, chunk_end in monthly_chunks(start, end):
        markets = await list_all_markets(client, series_ticker=series,
                                         min_close_ts=chunk_start, max_close_ts=chunk_end,
                                         max_pages=200)
```

Monthly chunks would cap hourly series at ~14,400 markets/month — well within the 40K limit.

### Alternative approaches:
- **Increase max_pages** — works but dramatically slows API fetching
- **Detect truncation and auto-chunk** — if pagination limit hit, re-fetch with halved date range
- **Warn the user** — at minimum, surface the truncation in the backtest output (currently only logged to stderr)

## Relationship to other issues
- #410 (closed) — added date filtering to API calls, but didn't solve the pagination cap for high-volume series
- #456 — concentration limits; accurate backtest data needed to validate these limits

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by jonnymobs907**
## Summary
- Split backtest market fetch into monthly chunks so high-volume series (KXINX, KXNASDAQ100 with ~480 markets/day) stay under the 40K pagination cap
- Deduplicate markets across chunk boundaries to prevent double-counting
- Surface truncation warnings in Rich console output and JSON results
- Added 8 tests for `monthly_chunks()` covering edge cases (leap year, year boundary, single day, last-day-of-month start)

Closes #459

## Test plan
- [x] All 969 unit tests pass
- [x] Lint clean on changed files
- [ ] Run `gimmes backtest --from 2026-01-01 --to 2026-04-01` — verify no "Pagination limit reached" warnings on stderr
- [ ] If truncation still occurs on any monthly chunk, verify the warning appears in the backtest output (not just stderr)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/gimmes/backtest/engine.py [hunk 1]
```diff
@@ -284,24 +314,47 @@ async def run_backtest(
...
+                    truncated_chunks.append(f"{series} ({label})")
+                    logger.warning(
+                        "Pagination limit hit for %s in %s (%d markets)",
+                        series, label, len(markets),
+                    )
+                all_markets.extend(markets)
+            except Exception:
+                logger.warning(
+                    "Failed to fetch series %s chunk", series, exc_info=True,
+                )
```
[the remark this answers is not in the document]
Comment (jonnymobs907) [reply src/gimmes/backtest/engine.py]: Fixed in b7db9d2. Error log now includes the month label for the failed chunk.

File: src/gimmes/backtest/engine.py [hunk 2]
```diff
@@ -284,24 +314,47 @@ async def run_backtest(
...
+
+    # Deduplicate markets that may appear in adjacent chunks
+    seen_tickers: set[str] = set()
+    deduped: list[Market] = []
+    for m in all_markets:
+        if m.ticker not in seen_tickers:
+            seen_tickers.add(m.ticker)
+            deduped.append(m)
+    all_markets = deduped
```
[the remark this answers is not in the document]
Comment (jonnymobs907) [reply src/gimmes/backtest/engine.py]: The `monthly_chunks()` pure function is thoroughly tested (8 cases). Mocking `list_all_markets()` for a full async integration test is heavy for this PR. The deduplication is a simple dict-based dedup that's hard to get wrong. The existing `original_by_ticker` dict on line 371 also deduplicates by ticker downstream.
